// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.password_manager.core.browser.proto";

option optimize_for = LITE_RUNTIME;

package password_manager;

import "chrome/browser/password_manager/android/protos/password_with_local_data.proto";
import "chrome/browser/password_manager/android/protos/password_info.proto";

// Response to a request sent to Google Mobile Services to request a list of
// affiliate passwords.
message ListAffiliatedPasswordsResult {
  // Password affiliated with the url specified in list affiliated passwords
  // request.
  message AffiliatedPassword {
    // Password with local data itself.
    optional PasswordWithLocalData password_data = 1;

    // Branding info for the password.
    optional PasswordInfo password_branding_info = 2;

    // Indicates whether sign on realm of this password is in the same
    // credential sharing affiliation as the url specified in list affiliated
    // passwords request.
    optional bool is_credential_sharing_affiliation_match = 3;

    // Indicates whether sign on realm of this password is in the same grouping
    // affiliation as the url specified in list affiliated passwords request.
    optional bool is_grouping_affiliation_match = 4;
  }

  // List of affiliated passwords. Only passwords with SCHEME_HTML
  // with https://, android:// http:// and federation:// sign on realms are
  // returned.
  repeated AffiliatedPassword affiliated_passwords = 1;
}
